hastory-server/src/Data/Hastory/Server/Handler/Users.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Handler.Users where
+
+import Data.Hastory.Server.Handler.Import
+
+createUserHandler :: UserForm -> HastoryHandler UserId
+createUserHandler userForm@UserForm {..} = do
+  mUser <- runDB . getBy $ UniqueUsername userFormUserName
+  if isNothing mUser && isValid userForm
+    then buildAndInsertUser userFormUserName userFormPassword
+    else throwError err400
+  where
+    buildAndInsertUser name password = do
+      user <- User name <$> liftIO (hashPassword . mkPassword $ password)
```

Comment:
Won't this be rather slow if we don't have a difficulty option for the hash?

Reply:
Good question. [The Docs](https://www.stackage.org/haddock/lts-16.2/password-2.0.1.1/Data-Password-Bcrypt.html) indicate that the default cost is `10`, and that valid costs are in the interval [4, 31]. Do you think this is a likely bottleneck? I'd be happy to change it, but I'll admit that that the need (and repercussions) of doing so elude me.

Reply:
@NorfairKing choosing 4 for the difficulty was tremendously helpful - the `hastory-server` tests went from over 10 seconds to under 1. Didn't realize how much of a difference that would make.